### PR TITLE
[WIP] Add LIST command to the health mgmt API and persist configuration

### LINKED ISSUE #3414: Simple way to disable alerts
I think it would be useful, if there was such option - through which, admin could disable any alert.
For example via template name for alert:

alerts to disable = *last_collected_secs inbound_packets_dropped *dropped_ratio

Of course glob should be supported as "for send charts matching" options for example.

### PR DESCRIPTION
**Pull request by cakrit**
Add a `LIST` command to show active silencers.
Automatically persist the silencers across restarts. At startup, load the saved configuration.

### CODE REVIEW

**Comment by cakrit** [comment]
This works now, but I need some help with these valgrind errors that I don't understand. Everything should be initialized, I'm missing something.

```
==27797== For counts of detected and suppressed errors, rerun with: -v
==27797== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
==27798== Conditional jump or move depends on uninitialised value(s)
==27798==    at 0x483AC78: strlen (vg_replace_strmem.c:460)
==27798==    by 0x1FF63A: json_walk_array (json.c:160)
==27798==    by 0x1FFB09: json_walk_object (json.c:220)
==27798==    by 0x1FFE62: json_walk_tree (json.c:260)
==27798==    by 0x1FFF25: json_parse (json.c:281)
==27798==    by 0x140170: health_silencers_init (health.c:168)
==27798==    by 0x14029E: health_init (health.c:182)
==27798==    by 0x156C79: rrd_init (rrdhost.c:448)
==27798==    by 0x122480: main (main.c:1087)
==27798==  Uninitialised value was created by a stack allocation
==27798==    at 0x1FFDA1: json_walk_tree (json.c:251)
==27798== 
==27798== Conditional jump or move depends on uninitialised value(s)
==27798==    at 0x483AC78: strlen (vg_replace_strmem.c:460)
==27798==    by 0x4C94886: vfprintf (in /usr/lib/libc-2.28.so)
==27798==    by 0x4CB54A6: vsprintf (in /usr/lib/libc-2.28.so)
==27798==    by 0x4C9B5C7: sprintf (in /usr/lib/libc-2.28.so)
==27798==    by 0x1FF6F2: json_walk_array (json.c:165)
==27798==    by 0x1FFB09: json_walk_object (json.c:220)
==27798==    by 0x1FFE62: json_walk_tree (json.c:260)
==27798==    by 0x1FFF25: json_parse (json.c:281)
==27798==    by 0x140170: health_silencers_init (health.c:168)
==27798==    by 0x14029E: health_init (health.c:182)
==27798==    by 0x156C79: rrd_init (rrdhost.c:448)
==27798==    by 0x122480: main (main.c:1087)
==27798==  Uninitialised value was created by a stack allocation
==27798==    at 0x1FFDA1: json_walk_tree (json.c:251)
```

File: libnetdata/json/json.c [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+	return start - init;
+}
+
+size_t json_walk_tree(char *js, jsmntok_t *t, void *callback_data, int (*callback_function)(struct json_entry *))
+{
+	JSON_ENTRY e;
+	strcpy(e.name, "");
+	strcpy(e.fullname, e.name);
+	e.callback_data = callback_data;
+	e.callback_function = callback_function;
```
Comment (ktsaou) [inline libnetdata/json/json.c]: The whole block above leaves `e` uninitialized.
To initialize it, use something like this:

```c
	JSON_ENTRY e = {
			.name = "",
			.fullname = "",
			.callback_data = "",
			.callback_function = callback_function
	};
```

This way, all members not specified, are automatically set to zero.

**Comment by ktsaou** [comment]
This PR has several compiler warnings.
Please compile netdata with this to reveal them:

```
CFLAGS="-O1 -ggdb -Wall -Wextra -DNETDATA_INTERNAL_CHECKS=1" ./netdata-installer.sh --disable-lto --dont-wait
```

File: libnetdata/json/json.c [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+
+size_t json_walk_array(char *js, jsmntok_t *t, size_t nest, size_t start, JSON_ENTRY *e)
+{
+	JSON_ENTRY ne;
+
+	char old = js[t[start].end];
+	js[t[start].end] = '\0';
+	ne.original_string = &js[t[start].start];
+
+	memcpy(&ne, e, sizeof(JSON_ENTRY));
```
Comment (ktsaou) [inline libnetdata/json/json.c]: `ne.oririnal` is overwritten with this `memcpy()`

File: libnetdata/json/json.c [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+	ne.type = JSON_ARRAY;
+	ne.data.items = t[start].size;
+	ne.callback_function = NULL;
+	strcpy(ne.name, "");
```
Comment (ktsaou) [inline libnetdata/json/json.c]: `ne.name[0] = '\0'` has the same effect and it is way faster.

File: libnetdata/json/json.c [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+	strcpy(ne.fullname, ne.name);
```
Comment (ktsaou) [inline libnetdata/json/json.c]: `ne.fullname[0] = '\0'` has the same effect and it is way faster.

File: libnetdata/json/json.c [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+}
+
+size_t json_walk_object(char *js, jsmntok_t *t, size_t nest, size_t start, JSON_ENTRY *e)
+{
+	JSON_ENTRY ne;
+
+	char old = js[t[start].end];
+	js[t[start].end] = '\0';
+	ne.original_string = &js[t[start].start];
+	strcpy(ne.name, "");
```
Comment (ktsaou) [inline libnetdata/json/json.c]: `ne.name[0] = '\0'` has the same effect and it is way faster.

File: libnetdata/json/json.c [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+	strcpy(ne.fullname, ne.name);
```
Comment (ktsaou) [inline libnetdata/json/json.c]: `ne.name[0] = '\0'` has the same effect and it is way faster.

File: libnetdata/json/json.c [hunk 7]
```diff
@@ -0,0 +1,294 @@
...
+	memcpy(&ne, e, sizeof(JSON_ENTRY));
```
Comment (ktsaou) [inline libnetdata/json/json.c]: whatever changed in `ne` before this, is lost with this `memcpy()`.

File: libnetdata/json/json.c [hunk 8]
```diff
@@ -0,0 +1,294 @@
...
+				break;
+		}
+	}
+	return start - init;
+}
+
+size_t json_walk_tree(char *js, jsmntok_t *t, void *callback_data, int (*callback_function)(struct json_entry *))
+{
+	JSON_ENTRY e;
+	strcpy(e.name, "");
```
Comment (ktsaou) [inline libnetdata/json/json.c]: `ne.name[0] = '\0'` has the same effect and it is way faster.

File: libnetdata/json/json.c [hunk 9]
```diff
@@ -0,0 +1,294 @@
...
+	strcpy(e.fullname, e.name);
```
Comment (ktsaou) [inline libnetdata/json/json.c]: `ne.fullname[0] = '\0'` has the same effect and it is way faster.

File: libnetdata/json/json.c (hunk 1 above)
Comment (cakrit) [reply libnetdata/json/json.c]: Done

File: libnetdata/json/json.c (hunk 3 above)
Comment (cakrit) [reply libnetdata/json/json.c]: Done

File: libnetdata/json/json.c (hunk 5 above)
Comment (cakrit) [reply libnetdata/json/json.c]: Done

File: libnetdata/json/json.c (hunk 6 above)
Comment (cakrit) [reply libnetdata/json/json.c]: Done

File: libnetdata/json/json.c (hunk 8 above)
Comment (cakrit) [reply libnetdata/json/json.c]: Done

File: libnetdata/json/json.c (hunk 9 above)
Comment (cakrit) [reply libnetdata/json/json.c]: Done

**Comment by cakrit** [comment]
Closing this, it will progress in #6274

**Pull request closed without merging**